filesmanager/tests/test_filesmanager.py

```diff
@@ -0,0 +1,949 @@
+        super().setUp()
+        self.request = Mock(
+            body=json.dumps({}).encode("utf-8"),
+            method="POST",
+            status_code_success=HTTPStatus.OK,
+        )
+
+    def test_get_directories_anonymous_user(self):
+        """
+        Check get directories JSON handler for anonymous user.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories list are returned without the unpublished directory.
+        """
+        self.xblock.get_current_user.return_value.opt_attrs.get.return_value = True
+        expected_result = {
+            "status": "success",
+            "contents": {
+                "id": "root",
+                "name": "Root",
+                "type": "directory",
+                "path": "Root",
+                "parentId": "",
+                "metadata": {},
+                "children": [],
+            },
+        }
+
+        response = self.xblock.get_directories(self.request)
+
+        self.xblock.fill_unpublished.assert_not_called()
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(expected_result, response.json)  # pylint: disable=no-member
+
+    def test_get_directories_non_anonymous_user(self):
+        """
+        Check get directories JSON handler for non-anonymous user.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories list are returned with the unpublished directory.
+        """
+        self.xblock.get_current_user.return_value.opt_attrs.get.return_value = None
+        expected_result = {
+            "status": "success",
+            "contents": self.xblock.directories,
+        }
+
+        response = self.xblock.get_directories(self.request)
+
+        self.xblock.fill_unpublished.assert_called()
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(expected_result, response.json)  # pylint: disable=no-member
+
+    def test_sync_content(self):
+        """
+        Check sync content JSON handler with files and directories.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories and files are synced.
+        """
+        content = {
+            "rootFolderId": "root",
+            "treeFolders": {
+                "id": "root",
+                "name": "Root",
+                "type": "directory",
+                "path": "Root",
+                "metadata": {},
+                "parentId": "",
+                "children": [
+                    {
+                        "id": "unpublished",
+                        "name": "Unpublished",
+                        "type": "directory",
+                        "path": "Root/Unpublished",
+                        "metadata": {},
+                        "parentId": "root",
+                        "children": [],
+                    }
+                ],
+            },
+        }
```

Comment:
This is what the default tree looks like. Can we modify it a bit more?

Reply:
Done